### PR TITLE
[layer] Enable bias hh in rnn, rnncell, lstm, zoneout lstm, grucell

### PR DESCRIPTION
**Pull request by lhs8928**
Commit 1:     [rnncell] enable 2 bias
    
     - Make a integrate bias property. It decide whether integrate 2 bias to 1
       or not. It will be used in rnn variant for now.
     - Added bias_hh in rnncell
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 2:     [rnn] enable 2 bias
    
     - Enable bias_hh in rnn
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 3:     [lstm, lstmcell, zoneout] enable 2 bias
    
     - Enable bias_hh in lstm, lstmcell, zoneout lstmcell layers
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

Commit 4:     [grucell] enable 2 bias
    
     - Enable bias_hh in grucell
    
    Self evaluation:
    
    Build test: [X]Passed [ ]Failed [ ]Skipped
    Run test: [X]Passed [ ]Failed [ ]Skipped
    
    Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

**Comment by zhoonit** [comment]
fyi) first commit seems to clashing with https://github.com/nnstreamer/nntrainer/pull/1780/commits/3e9afea4d44fe7bee3aa60ff35a31aa1a4841ec9

File: nntrainer/layers/rnncell.cpp [hunk 1]
```diff
@@ -47,69 +59,90 @@ void RNNCellLayer::finalize(InitLayerContext &context) {
...
 
   if (context.getNumInputs() != 1) {
     throw std::invalid_argument("RNNCell layer takes only one input");
   }
 
   // input_dim = [ batch, 1, 1, feature_size ]
-  const TensorDim &input_dim = context.getInputDimensions()[0];
-  if (input_dim.channel() != 1 && input_dim.height() != 1) {
+  const TensorDim &input_dim = context.getInputDimensions()[SINGLE_INOUT_IDX];
+  if (input_dim.channel() != 1 || input_dim.height() != 1) {
```
Comment (kparichay) [inline nntrainer/layers/rnncell.cpp:80]: :100:

File: nntrainer/layers/rnncell.cpp [hunk 2]
```diff
@@ -129,131 +163,205 @@ void RNNCellLayer::exportTo(Exporter &exporter,
...
 
-  Tensor &input = context.getInput(SINGLE_INOUT_IDX);
-  const TensorDim &input_dim = input.getDim();
-  const unsigned int batch_size = input_dim[0];
   Tensor &hidden_states =
     context.getTensor(wt_idx[RNNCellParams::hidden_state]);
   hidden_states.reshape({max_timestep, 1, batch_size, unit});
+  Tensor prev_hidden_state;
+  if (!timestep) {
+    prev_hidden_state = Tensor(batch_size, unit);
```
Comment (kparichay) [inline nntrainer/layers/rnncell.cpp:199]: why change from the existing version and add new tensor creation + zeroing out?

File: nntrainer/layers/rnncell.cpp [hunk 3]
```diff
@@ -129,131 +163,205 @@ void RNNCellLayer::exportTo(Exporter &exporter,
...
+  Tensor &djdbias_hh = !disable_bias && !integrate_bias
+                         ? context.getWeightGrad(wt_idx[RNNCellParams::bias_hh])
+                         : empty;
 
   Tensor &hidden_states =
     context.getTensor(wt_idx[RNNCellParams::hidden_state]);
   hidden_states.reshape({max_timestep, 1, batch_size, unit});
+  Tensor prev_hidden_state;
+  if (!timestep) {
+    prev_hidden_state = Tensor(batch_size, unit);
```
Comment (kparichay) [inline nntrainer/layers/rnncell.cpp:288]: same here.

**Review by kparichay — APPROVED** [review]
Overall LGTM other than the minor comment

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Review by zhoonit — APPROVED** [review]
LGTM

**Pull request merged**